### PR TITLE
Support signing deploy account transactions

### LINKED ISSUE #421: Support signing `DEPLOY_ACCOUNT` transactions
Methods should be added to AccountClient and to Signer interface, to allow signing `DEPLOY_ACCOUNT` transactions.

### PR DESCRIPTION
**Pull request by cptartur**
### **Please check if the PR fulfills these requirements**
- [ ] The formatter, linter, and tests all run without an error
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

Closes #421 

This PR doesn't include test for this feature, they will need to be added in another PR

### CODE REVIEW

File: starknet_py/net/account/account_client.py [hunk 1]
```diff
@@ -434,6 +435,47 @@ async def sign_declare_transaction(
...
+            and used to calculate deployed contract address
+        :param max_fee: Max fee to be paid for deploying account transaction. Enough tokens must be prefunded before
+            sending the transaction for it to succeed.
+        :return: Signed DeployAccount transaction
+        """
+        # TODO not sure if this should be added, but the example transaction from starkware uses version 1
+        if self.supported_tx_version != 1:
+            raise ValueError(
+                "Signing deploy account transactions is only supported with transaction version 1"
+            )
```
Comment (war-in) [inline starknet_py/net/account/account_client.py:461]: I've checked that and this should be added
image

**Comment by cptartur** [comment]
`DECLARE` transactions are currently not covered by unit tests. Referenced in #433

File: starknet_py/net/account/account_client.py [hunk 2]
```diff
@@ -434,6 +435,46 @@ async def sign_declare_transaction(
 
         return dataclasses.replace(declare_tx, signature=signature, max_fee=max_fee)
 
+    async def sign_deploy_account_transaction(
+        self,
+        *,
+        class_hash: int,
+        contract_address_salt: int,
+        constructor_calldata: Optional[List[int]] = None,
+        max_fee: int,
```
Comment (war-in) [inline starknet_py/net/account/account_client.py:444]: What about `auto_estimate` parameter? If we can't do it now please add `TODO`

File: starknet_py/net/signer/test_stark_curve_signer.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            contract_address_salt=0x2,
+            constructor_calldata=[1, 2, 3, 4],
+            max_fee=10000,
+            signature=[],
+            nonce=23,
+            version=1,
+        ),
+    ],
+)
+def test_sign_invoke_transaction(transaction):
```
Comment (war-in) [inline starknet_py/net/signer/test_stark_curve_signer.py:30]: `test_sign_transaction`?

**Review by war-in — APPROVED** [review]

**Pull request merged**